### DISCUSSION TITLE
Help in understanding of graph_sage_unsup_ppi correctly?

### DISCUSSION THREAD

**Discussion by lisiq** [pyg-team/pytorch_geometric#6762] [Q&A] [answered]
Hi everyone! I want to make sure that I get [this](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/graph_sage_unsup_ppi.py) model properly:)

1. LinkNeighborLoader will return subgraphs with positive and negative edges, where we sample 10 neighbours for each node and edge type for 10 iterations. Since we have grouped all training graphs into a single graph, does this mean that we have created also negative edges between different graphs?
2. How many subgraphs are created?
3. We train the model with the link prediction idea?
4. Why don't we need a big number for epochs?
5. How is the aggregation done here after one epoch is finished for all the mini-batches?
6. Is there a paper where this is based from, or a paper that uses this dataset in a similar way?

I'm sorry for all the questions, and thank you in advance!

**Comment by wsad1** [comment, marked as answer]
Thanks for your questions
1. Firstly PPI is a homogenous graph so there's only a single node and edge type. The `LinkNeighborLoader` will sample 10 neigbhors 2-hops away that's what `num_neighbors = [10, 10]` indicates.
2. `LinkNeibhborLoader` will created around `num_edges`/2048 subgraphs, each subgraph is used in a batch
3. Yes
4. The code is really to illustrate how to do unsupervised training, also I think the original paper which you can find [here](https://github.com/pyg-team/pytorch_geometric/pull/4416) used 10 epochs
5. Which aggregation are you referring to here?
6. Yes mentioned that in 4.

**Reply by lisiq** [reply]
Thank you very much for the answer!
Question 5 doesn't make sense sorry.
I am still a bit confused why with only 10 epochs it performs so well in a complex task. In my experiments also only 1 epoch is enough for good predictions 😅
